### PR TITLE
Handle no internet connection

### PR DESCRIPTION
**Pull request by rob-horizontal-iOS**
For more information on contributing to Tella source code, see the [Contributor Guidelines](contributing/contributor_guide.md).

## Type of change

**Description:**

**Select the type of change(s) made in this pull request:**
- [x] Bug fix *(Fixes an issue)*
- [ ] New feature *(Adds functionality)*
- [ ] Documentation *(Fix to documentation)*

----------------------------------------------------------------------------------------

Fixes #issue-number

## Proposed changes 

*
*
*

### CODE REVIEW

File: Tella/Data/Networking/APICall/APIError.swift [hunk 1]
```diff
@@ -20,6 +21,8 @@ extension APIError: LocalizedError {
             return customErrorMessage(errorCode: code)
         case .unexpectedResponse:
             return "Unexpected response from the server"
+        case .noInternetConnection:
+            return "No Internet connection. Try again when you are connected to the Internet."
```
Comment (ValbuenaG) [inline Tella/Data/Networking/APICall/APIError.swift:25]: should we add translations for this?

cc @dhekra-rouatbi

**Review by ValbuenaG — APPROVED** [review]
Approved but let's add translations if we think it's necessary

File: Tella/Data/Networking/APICall/APIError.swift (hunk 1 above)
Comment (rob-horizontal-iOS) [reply Tella/Data/Networking/APICall/APIError.swift]: @ValbuenaG  added the translation

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Data/Networking/APICall/URLRequest.swift [hunk 2]
```diff
@@ -15,6 +15,10 @@ extension WebRepository {
     func getAPIResponse<Value>(endpoint: any APIRequest) -> APIResponse<Value>
     where Value: Decodable {
         do {
+            guard (NetworkMonitor.shared.isConnected) else {
+                return Fail<(Value,[AnyHashable:Any]?), APIError>(error: APIError.noInternetConnection)
```
[the remark this answers is not in the document]
Comment (rob-horizontal-iOS) [reply Tella/Data/Networking/APICall/URLRequest.swift]: Replaced for no internet connection and invalidURL

File: Tella/Data/MainAppModel.swift [hunk 3]
```diff
@@ -76,6 +76,8 @@ class MainAppModel: ObservableObject, AppModelFileManagerProtocol {
     
     init(networkMonitor:NetworkMonitor) {
         self.networkMonitor = networkMonitor
+        // To initialize the NetworkMonitor singleton class
+        _ = NetworkMonitor.shared.isConnected
```
Comment (dhekra-rouatbi) [inline Tella/Data/MainAppModel.swift:80]: You don't need to reinitiate the `NetworkMonitor` here, you can update it in `AppViewState` 
`   let networkMonitor = NetworkMonitor.shared.    
`

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Scenes/Settings/ViewModel/UwaziServerViewModel.swift [hunk 4]
```diff
@@ -176,7 +182,6 @@ class UwaziServerViewModel: ObservableObject {
 
     // MARK: - Check URL API Call Methods
     func checkURL() {
-        isLoading = true
```
Comment (dhekra-rouatbi) [inline Tella/Scenes/Settings/ViewModel/UwaziServerViewModel.swift:179]: If we remove the loader how we can tell the user to wait

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Data/MainAppModel.swift (hunk 3 above)
Comment (rob-horizontal-iOS) [reply Tella/Data/MainAppModel.swift:80]: Code moved to `AppViewState `

**Pull request merged**